src/Symfony/Component/Security/Core/User/User.php

```diff
@@ -43,23 +43,23 @@ public function __construct(?string $username, ?string $password, array $roles =
         $this->roles = $roles;
     }
 
-    public function __toString()
+    public function __toString(): string
     {
         return $this->getUsername();
     }
 
     /**
      * {@inheritdoc}
      */
-    public function getRoles()
+    public function getRoles(): array
```

Comment:
I was not sure if we wanted to type this class.
The properties are all typed in the constructor, but it might be bypassed.
WDYT @nicolas-grekas ?

Reply:
returning an array is the only "legal" return type, right? if confirmed, this is good, bypassing the type would violate the contract, isn't it?